src/Native/Unix/System.Security.Cryptography.Native/pal_ecc_import_export.cpp

```diff
@@ -108,8 +108,16 @@ extern "C" int32_t CryptoNative_GetECKeyParameters(
 
     if (includePrivate)
     {
-        *d = const_cast<BIGNUM*>(EC_KEY_get0_private_key(key));
-        *cbD = BN_num_bytes(*d);
+        const BIGNUM* const_bignum_privateKey = EC_KEY_get0_private_key(key);
+        if (const_bignum_privateKey)
+        {
+            *d = const_cast<BIGNUM*>(const_bignum_privateKey);
+            *cbD = BN_num_bytes(*d);
+        }
+        else
+        {
+            goto error;
```

Comment:
Does this result in an exception indicating that there's no private key?  If the text isn't very helpful then this should probably just change to 

```c++
*d = nullptr;
*cbD = 0;
```

and let the existing check handle it.

Reply:
I wanted to keep the existing error structure so we get the freeing and cleanup without any code duplication.

I modified things a bit to keep using `goto error` but with a different return code that indicates failure due to this particular problem. That way we can check the RC in the interop code and throw a CryptographicException with the right error code.